src/http-client.ts

```diff
@@ -141,10 +141,17 @@ export class HttpClient {
       })
       .then(result => {
         if (Request.prototype.isPrototypeOf(result)) {
-          return this.fetch(result);
+          let recursiveFetch = this.fetch(result);
+          trackRequestEnd(this);
+          return recursiveFetch;
         }
+        return Promise.resolve(result);
+      }).then((result) => {
```

Comment:
I think it would better be:

```ts
.then(
  result => {
    trackRequestEnd(this);
    return result
  },
  error => {
    trackRequestEnd(this);
    throw / promise reject ... error
  }
)
```